### ISSUE TITLE
pip install banditpam error

### ISSUE THREAD

**Issue by KeyCode17** [motiwari/BanditPAM#263]
LINK : warning LNK4044: unrecognized option '/v'; ignored
LINK : fatal error LNK1181: cannot open input file 'libopenblas.lib'
error: command 
'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX86\\x86\\link.exe' failed with exit code 1181
 i got output like that if using:
      
     pip install banditpam

when i try 

     use pip install .

after 

     mkdir build && cd build && cmake -DCARMA_INSTALL_LIB=ON .. && sudo cmake --build . --config Release --target

installed on headers/carma i got error like this
cl : Command line warning D9002 : ignoring unknown option '-O3'
build_medoids_python.cpp
src\python_bindings\build_medoids_python.cpp(11): fatal error C1083: Cannot open include file: 'carma': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX86\\x86\\cl.exe' failed with exit code 2
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for banditpam
Failed to build banditpam
ERROR: Could not build wheels for banditpam, which is required to install pyproject.toml-based projects

Please help me

**Comment by Adarsh321123** [comment]
Hi @KeyCode17, thanks for filing an issue!
1. `pip install banditpam` is currently unsupported on Windows. `v4.0.4` will upload BanditPAM Windows wheels to PyPI.
2. I think you're getting a carma issue because you might not have a carma subdirectory (as in, there might not be any code under the `headers/carma` directory). Can you try removing `headers/carma`, cloning carma, and then running `mkdir build && cd build && cmake -DCARMA_INSTALL_LIB=ON .. && sudo cmake --build . --config Release --target`?

**Comment by KeyCode17** [comment]
which carma i need to clone ?

**Comment by Adarsh321123** [comment]
Just clone the most recent version of carma from [its repo](https://github.com/RUrlus/carma) using `git clone https://github.com/RUrlus/carma.git`!

**Comment by KeyCode17** [comment]
do i run this inside carma ? `mkdir build && cd build && cmake -DCARMA_INSTALL_LIB=ON .. && sudo cmake --build . --config Release --target`

**Comment by Adarsh321123** [comment]
Yes, after cloning you should have a `headers/carma` directory. `cd` into it and then run `mkdir build`, `cd build`, `cmake -DCARMA_INSTALL_LIB=ON ..`, and `cmake --build . --config Release --target install` (you can run this last command as administrator since `sudo` doesn't exist on Windows).

**Comment by KeyCode17** [comment]
Now error like this

build\lib.win32-cpython-310\banditpam.cp310-win_amd64.pyd : fatal error LNK1120: 132 unresolved externals
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX86\\x86\\link.exe' failed with exit code 1120
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for banditpam
Failed to build banditpam
ERROR: Could not build wheels for banditpam, which is required to install pyproject.toml-based projects

**Comment by Adarsh321123** [comment]
Have you followed these steps from [install_windows.md](https://github.com/motiwari/BanditPAM/blob/main/docs/install_windows.md)?:

```
 1) Add the location of `cl.exe` to PATH in Environment Variables (e.g. `C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64`).
 2) Run `python -m pip install .` in the home directory (`/BanditPAM`)
 2) Add the file `C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\Hostx86\x64\clang_rt.asan_dynamic-x86_64.dll` to `build\lib.win-amd64-cpython-310`
 3) Run `python -m pip install .` in the home directory (`/BanditPAM`)
```

**Comment by KeyCode17** [comment]
i can't find build\lib.win-amd64-cpython-310

**Comment by Adarsh321123** [comment]
Can you please show me the contents of your `build` directory?

**Comment by Allan0820** [comment]
@KeyCode17 did you resolve the issue?
@Adarsh321123 can this issue be closed

**Comment by Allan0820** [comment]
Hi @motiwari im running a sweep check. Please close the issue if you feel that it has been solved/ is not important and can be left in stalemate.

**Comment by motiwari** [comment]
@Allan0820 appreciate the sweep! I believe this issue has been resolved. If not, we can address it when re-filed.
